core/src/main/java/org/apache/calcite/rel/metadata/RelMdMaxRowCount.java

```diff
@@ -104,6 +105,10 @@ public class RelMdMaxRowCount
     return mq.getMaxRowCount(rel.getInput());
   }
 
+  public @Nullable Double getMaxRowCount(Sample rel, RelMetadataQuery mq) {
+    return mq.getMaxRowCount(rel.getInput());
```

Comment:
Hi,I have a question is that why getMaxRowCount method not consider the sample rate?

Reply:
Bernoulli sampling can, in the worst case, return all rows, or no rows.

Not sure about system sampling.